front/src/pages/settings/data-model/SettingsObjectDetail.tsx

```diff
@@ -34,34 +31,46 @@ export const SettingsObjectDetail = () => {
   const navigate = useNavigate();
 
   const { pluralObjectName = '' } = useParams();
-  const activeObject = activeObjectItems.find(
-    (activeObject) => activeObject.name.toLowerCase() === pluralObjectName,
+  const { activeObjects } = useObjectMetadata();
+  const activeObject = activeObjects.find(
+    (activeObject) => activeObject.namePlural === pluralObjectName,
   );
 
   useEffect(() => {
-    if (!activeObject) navigate(AppPath.NotFound);
-  }, [activeObject, navigate]);
+    if (activeObjects.length && !activeObject) {
+      navigate(AppPath.NotFound);
+    }
+  }, [activeObject, activeObjects.length, navigate]);
+
+  const { icons } = useLazyLoadIcons();
+
+  const activeFields = activeObject?.fields.filter(
+    (fieldItem) => fieldItem.isActive,
+  );
+  const disabledFields = activeObject?.fields.filter(
+    (fieldItem) => !fieldItem.isActive,
+  );
 
   return (
     <SubMenuTopBarContainer Icon={IconSettings} title="Settings">
       <SettingsPageContainer>
         <Breadcrumb
           links={[
             { children: 'Objects', href: '/settings/objects' },
-            { children: activeObject?.name ?? '' },
+            { children: activeObject?.labelPlural ?? '' },
           ]}
         />
         {activeObject && (
           <SettingsAboutSection
-            Icon={activeObject?.Icon}
-            name={activeObject.name}
-            type={activeObject.type}
+            Icon={icons[activeObject?.icon || '']}
```

Comment:
Here we will be all the icons, it would be great to only load the required one.
Could you try to propose a loadIcon function as part as useLazyLoadIcons hook that would only load he desired icon?
https://stackoverflow.com/questions/61339259/how-to-dynamically-import-svg-and-render-it-inline/61472427#61472427
Not sure it's possible or it works

Reply:
I added a `useLazyLoadIcon` hook that only imports the required icon.